server/routerlicious/packages/services-core/src/runWithRetry.ts

```diff
@@ -108,6 +108,10 @@ export async function runWithRetry<T>(
 		}
 	}
 
+	if (shouldIgnoreError !== undefined && shouldIgnoreError(latestResultError) === true) {
+		return undefined as unknown as T; // Ensure a value of type T is returned
```

Comment:
May be add a warning saying `return undefined because error ignored`?

Reply:
A log is already written on line 75 above, I think that should be sufficient?

Reply:
I considered that but in that case `finally` block wont be executed for the metric telemetry. Thats why I added it at the end. This was a recent regression because of new code added on lines 115-121.

Reply:
I am pretty sure `finally` will be executed even if there is an early return or error throw:
![image](https://github.com/user-attachments/assets/dba476f6-fe04-4fa9-be1e-2c11180dcbde)

Reply:
ah ok, thanks for verifying @zhangxin511 . I will update it in that case.